### PR TITLE
Apps Layer Type Modification

### PR DESCRIPTION
**Pull request by yacyam**
Egos.h added with typedef of integers as i32, unsigned integers as u32.

All types in Apps Layer were modified to include i32 type, with u32 used at certain points.

Types that seemed clear to be made unsigned were changed to u32, but majority were kept as i32 to keep first request simple.

### CODE REVIEW

File: apps/system/sys_dir.c [hunk 1]
```diff
@@ -13,19 +13,19 @@
 #include <stdlib.h>
 
 /* To understand directory management, read tools/mkfs.c */
-int dir_do_lookup(int dir_ino, char* name) {
+i32 dir_do_lookup(i32 dir_ino, char* name) {
```
Comment (yhzhang0128) [inline apps/system/sys_dir.c:16]: Maybe we should make inode numbers u32?

File: apps/system/sys_proc.c [hunk 2]
```diff
@@ -12,14 +12,14 @@
 #include "disk.h"
 #include <string.h>
 
-static int app_ino, app_pid;
-static void sys_spawn(int base);
-static int app_spawn(struct proc_request *req);
+static i32 app_ino, app_pid;
```
Comment (yhzhang0128) [inline apps/system/sys_proc.c:15]: Similar here, maybe u32? But let me know if you have good reasons to use i32. I am just asking, not suggesting.

File: apps/system/sys_dir.c (hunk 1 above)
Comment (yhzhang0128) [reply apps/system/sys_dir.c:16]: OK, it seems that file descriptors are int type, so let's keep int.

**Comment by yhzhang0128** [comment]
Good job! And thank you for your pull request. I think about it and it seems better to use `int/uint` instead of `i32/u32`. Please update the PR accordingly. Below are some guidelines of where to use `int` and where to use `unit`.

### Using int:
1. inode numbers
2. return value of functions
3. the argc parameter of main()

### Using unsigned int (uint):
1. return value of sizeof() and strlen()
2. addresses and offsets
* the "off" parameter of app_read() in sys_proc.c
* the "base" parameter of sys_spawn() in sys_proc.c
* the "block_no" parameter of sys_proc_read() in sys_proc.c
3. counters starting from 0
* idx and nargs in sys_shell.c
* for loop variables that should be non-negative

**Comment by yacyam** [comment]
I changed all of the types to use the Int and UInt types, and have added uint to addresses and offsets, which mostly came through in sys_proc.c. Hopefully everything looks alright, and thank you for all the help.

**Pull request merged**